twoliter/embedded/rpm2img

```diff
@@ -688,51 +611,60 @@ mkfs_data() {
 #
 # If the other partition is available at runtime, the filesystem will be created
 # during first boot instead, providing flexibility at the cost of a minor delay.
-case "${PARTITION_PLAN}" in
-  unified)
-    mkfs_data "${OS_IMAGE}" "${partsize["DATA-A"]}M" "${partoff["DATA-A"]}"
+if [ "${XFS_DATA_PARTITION}" == "yes" ]; then
+  data_fs="xfs"
+else
+  data_fs="ext4"
+fi
+
+case "${PARTITION_PLAN}-${data_fs}" in
+  unified-ext4)
+    mkfs_data_ext4 "${OS_IMAGE}" "${partsize["DATA-A"]}M" "${partoff["DATA-A"]}" \
+      "${BOTTLEROCKET_DATA}" "${DATA_MOUNT}" "${UNLABELED}"
     ;;
-  split)
-    mkfs_data "${DATA_IMAGE}" "${partsize["DATA-B"]}M" "${partoff["DATA-B"]}"
+  unified-xfs)
+    mkfs_data_xfs "${OS_IMAGE}" "${partsize["DATA-A"]}M" "${partoff["DATA-A"]}" \
+      "${BOTTLEROCKET_DATA}"
+    ;;
+  split-ext4)
+    mkfs_data_ext4 "${DATA_IMAGE}" "${partsize["DATA-B"]}M" "${partoff["DATA-B"]}" \
+      "${BOTTLEROCKET_DATA}" "${DATA_MOUNT}" "${UNLABELED}"
+    ;;
```

Comment:
If you refactor `mkfs_data_xfs` and `mkfs_data_ext4` to take the same number of arguments, you can keep this closer to the previous code:

```
if [ "${XFS_DATA_PARTITION}" == "yes" ]; then
  data_fs_fn="mkfs_data_xfs"
else
  data_fs_fn="mkfs_data_ext4"
fi

case "${PARTITION_PLAN}" in
  unified)
    "${mkfs_data_fn}" "${OS_IMAGE}" "${partsize["DATA-A"]}M" "${partoff["DATA-A"]}" \
      "${BOTTLEROCKET_DATA}" "${DATA_MOUNT}" "${UNLABELED}"
    ;;
  split)
    "${mkfs_data_fn}" "${DATA_IMAGE}" "${partsize["DATA-B"]}M" "${partoff["DATA-B"]}" \
      "${BOTTLEROCKET_DATA}" "${DATA_MOUNT}" "${UNLABELED}"
    ;;
esac
```